json_serializable/lib/src/utils.dart

```diff
@@ -11,6 +11,42 @@ import 'package:analyzer/src/dart/resolver/inheritance_manager.dart'
 
 import 'package:source_gen/source_gen.dart';
 
+// TODO: still need handle tripe singe/double quotes within `value`
+String escapeDartString(String value) {
+  if (value.contains('\n')) {
+    return "r'''\n$value'''";
+  }
+
+  var containsSingleQuote = value.contains("'");
+  var contains$ = value.contains(r'$');
+
+  if (containsSingleQuote) {
+    if (value.contains('"')) {
+      // `value` contains both single and double quotes as well as `$`.
```

Comment:
we don't know if it contains `$` here. Are we assuming that it could so may as well escape it?

Reply:
Oops! Will dig in a bit...